src/libs/SignatureChecker.sol

```diff
@@ -121,13 +162,35 @@ library SignatureChecker {
         return recoverECDSASigner(digest, v, r, s);
     }
 
+    /**
+     * @dev    Returns the signer of an ECDSA/secp256k1 short signature for some digest.
+     * @dev    See https://eips.ethereum.org/EIPS/eip-2098
+     * @param  digest The hash of the data that was signed.
+     * @param  r      An ECDSA/secp256k1 signature parameter.
+     * @param  vs     An ECDSA/secp256k1 short signature parameter.
+     * @return error  An error, if any, that occurred during the signer recovery.
+     * @return signer The address of the account recovered form the signature (0 if error).
+     */
+    function recoverECDSASigner(
+        bytes32 digest,
+        bytes32 r,
+        bytes32 vs
+    ) internal pure returns (Error error, address signer) {
+        unchecked {
+            bytes32 s = vs & bytes32(0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff);
+            // We do not check for an overflow here since the shift operation results in 0 or 1.
+            uint8 v = uint8((uint256(vs) >> 255) + 27);
+            (error, signer) = recoverECDSASigner(digest, v, r, s);
```

Comment:
```suggestion
            return recoverECDSASigner(digest, v, r, s);
```

Reply:
Fixed in this commit: https://github.com/MZero-Labs/common/pull/8/commits/fcb4ccb4fe10e6a5002111e1e3766c056b6bfbec